### ISSUE TITLE
Add `imageGCThresholdPercent` support to `kubeletConfiguration`

### ISSUE THREAD

**Issue by Swalloow** [aws/karpenter#3110]
### Tell us about your request

- Adds `imageGCHighThresholdPercent`, `imageGCLowThresholdPercent` support into kubeletConfiguration
```
apiVersion: karpenter.sh/v1alpha5
kind: Provisioner
metadata:
  name: default
spec:
  ...
  kubeletConfiguration:
    imageGCHighThresholdPercent: 80
    imageGCLowThresholdPercent: 60
```

### Tell us about the problem you're trying to solve. What are you trying to do, and why is it hard?

I'm trying to configure Amazon EKS worker nodes to clean up the image cache at a specified percent of disk usage.
Ref: https://aws.amazon.com/premiumsupport/knowledge-center/eks-worker-nodes-image-cache/

### Are you currently working around this issue?

Currently I'm solving it with the userdata script.
Ref: https://aws.amazon.com/premiumsupport/knowledge-center/eks-worker-nodes-image-cache/

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by njtran** [comment]
Hey @Swalloow, thanks for the feature request. Are you willing to implement this? We can help guide you through the code base if you'd like to. For reference here are some places this would need to change: 

1) Add to Options API Definition - https://github.com/aws/karpenter-core/blob/913efba5245eac336630a291427c652bb51cce7b/pkg/apis/v1alpha5/provisioner.go#L115
2) Add into EKS Bootstrap script - https://github.com/aws/karpenter/blob/main/pkg/cloudprovider/amifamily/bootstrap/eksbootstrap.go#L52
3) Add into Bottlerocket config struct - https://github.com/aws/karpenter/blob/main/pkg/cloudprovider/amifamily/bootstrap/bottlerocketsettings.go#L45
4) Add into Bottlerocket Bootstrap script - https://github.com/aws/karpenter/blob/main/pkg/cloudprovider/amifamily/bootstrap/bottlerocket.go#L32

Feel free to ping me for any questions!

**Comment by Swalloow** [comment]
@njtran I'll try it. Thanks!

**Comment by njtran** [comment]
Thanks, feel free to ask for any questions on slack. I'll be around for the holidays.

**Comment by lucass4** [comment]
If @Swalloow is not able to work on this one, I would like to take this one @njtran

**Comment by digiserg** [comment]
I'm also interested in this feature and happy to contribute code.

**Comment by Swalloow** [comment]
I'm not able to work right now. Anyone can take it.

**Comment by digiserg** [comment]
> @njtran, I will work on this one since Swalloow cannot work on it. 🙇‍♂️

Thanks! Shout if you need a tester

**Comment by snorlaX-sleeps** [comment]
Sorry to necro this, would the team be open to adding more kubelet-config options in this manner?
We are currently setting several via userdata when we would have historically passed them as flags.
Could potentially open a PR for it

**Comment by ellistarn** [comment]
Sure thing. Would love to hear the use case and open minded to adding more flags.

**Comment by aRobinson-R7** [comment]
We currently have teams using both [cpu-management policies](https://kubernetes.io/docs/tasks/administer-cluster/cpu-management-policies/) and [memory-management policies](https://kubernetes.io/blog/2021/08/11/kubernetes-1-22-feature-memory-manager-moves-to-beta/) when configuring certain node groups (AWS Node Template + Provisioner) in clusters (both options include a feature-flag as well as configuration).

Historically (when using EC2 Launch Templates instead of AWS Node Templates) we would have set these options as kubelet-args in the user-data when starting the node.

The current workaround with AWS Node Template is to use `jq` on node start to insert these args into the `/etc/kubernetes/kubelet/kubelet-config.json`.
This isn't an issue (I guess), more of a shift away from being able to add any kubelet-arg to a specific provisioner / launch-template and having to support additional options as they become available in userdata.
Having written that I guess it's more a question of if Karpenter could support this dynamically (as options become available) rather than the end-user having to update their (probably shared) user-data template for each arg they want to add.

I should probably open a separate "question / discussion" issue around this rather than jumping on this thread, apologies.

**Comment by snorlaX-sleeps** [comment]
We currently have teams using both [cpu-management policies](https://kubernetes.io/docs/tasks/administer-cluster/cpu-management-policies/) and [memory-management policies](https://kubernetes.io/blog/2021/08/11/kubernetes-1-22-feature-memory-manager-moves-to-beta/) when configuring certain node groups (AWS Node Template + Provisioner) in clusters (both options include a feature-flag as well as configuration).

Historically (when using EC2 Launch Templates instead of AWS Node Templates) we would have set these options as kubelet-args in the user-data when starting the node.

The current workaround with AWS Node Template is to use `jq` on node start to insert these args into the `/etc/kubernetes/kubelet/kubelet-config.json`.
This isn't an issue (I guess), more of a shift away from being able to add any kubelet-arg to a specific provisioner / launch-template and having to support additional options as they become available in userdata.
Having written that I guess it's more a question of if Karpenter could support this dynamically (as options become available) rather than the end-user having to update their (probably shared) user-data template for each arg they want to add.

I should probably open a separate "question / discussion" issue around this rather than jumping on this thread, apologies.

**Comment by ygaurangj** [comment]
Hello,
I have added the Kubelet config arguments into my provisioner. but it is not working. I checked this by pulling dummy android images and disk usage goes above 80% but it wasn't delete anything.

kubeletConfiguration:
    imageGCHighThresholdPercent: 80
    imageGCLowThresholdPercent: 60

I have added above lines. Also when I check the kube-config.json there also I haven't found this arguments. 
well I am using EKS(1.24), Karpenter(0.27.0).
